pkg/rancher-desktop/config/settings.ts

```diff
@@ -475,113 +474,163 @@ function updateSettings(settings: Settings) {
+  profiles.locked = validateDeploymentProfile(profiles.locked, defaultSettings) ?? {};
 
-function readRegistryDeploymentProfile() {
-  throw new Error('INTERNAL ERROR: readRegistryDeploymentProfile not yet implemented.');
+  return profiles;
 }
 
-function validateDeploymentProfile(profile: any, schema: any) {
-  if (typeof profile === 'undefined') {
-    return undefined;
-  }
+/**
+ * Read and parse deployment profile files.
+ * @param rootPath the system or user directory containing profiles.
+ * @param defaultsPath the file path to the 'defaults' file.
+ * @param lockedPath the file path to the 'locked' file.
+ * @param parser the parser (JSON or plist) for parsing the files read.
+ * @returns the defaults and/or locked objects if they exist, or
+ *          throws an exception if there is an error parsing the locked file.
+ */
+function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: string, parser: any) {
+  let defaults;
+  let locked;
 
-  Object.keys(profile).forEach((key) => {
-    if (key in schema) {
-      if (typeof profile[key] === typeof schema[key]) {
-        if (typeof profile[key] === 'object') {
-          validateDeploymentProfile(profile[key], schema[key]);
-        }
-      } else {
-        console.log(`Deployment Profile ignoring '${ key }'. Wrong type.`);
-        delete profile[key];
-      }
-    } else {
-      console.log(`Deployment Profile ignoring '${ key }'. Not in schema.`);
-      delete profile[key];
+  try {
+    const defaultsData = fs.readFileSync(join(rootPath, defaultsPath), 'utf8');
+
+    defaults = parser.parse(defaultsData);
+  } catch {}
+  try {
+    const lockedData = fs.readFileSync(join(rootPath, lockedPath), 'utf8');
+
+    locked = parser.parse(lockedData);
+  } catch (ex: any) {
+    if (ex.code !== 'ENOENT') {
+      throw new Error(`Error parsing locked deployment profile: ${ ex }`);
     }
-  });
+  }
 
-  return profile;
+  return { defaults, locked };
 }
 
 /**
- * Merge profile into config and return result.
- * @param {*} config The config settings file.
- * @param {*} profile The deployment profile.
- * @param {*} lockedOnly Only overwrite settings for locked profile values.
- * @return {*} The merged configuration file.
+ * Windows only. Read settings values from registry using schemaObj as a template.
+ * @param object null - used for recursion.
+ * @param schemaObj the object used as a template for navigating registry.
+ * @param regKey the registry key obtained from nativeReg.openKey().
+ * @param regPath the path to the object relative to regKey.
+ * @returns undefined, or the registry data as an object.
  */
-export function mergeDeploymentProfile(config: any, profile: any, lockedOnly: boolean): any {
-  if (typeof config === 'undefined' || typeof profile === 'undefined') {
-    return (profile || config);
+function readRegistryUsingSchema(object: any, schemaObj: any, regKey: nativeReg.HKEY, regPath: string[]): any {
```

Comment:
`object ` isn't used, and is only passed as `null`

Reply:
Removed the unused parm.